templates/CRM/Financial/Form/BatchTransaction.tpl

```diff
@@ -238,12 +242,14 @@ function buildTransactionSelectorAssign(filterSearch) {
     CRM.$.ajax({
       "dataType": 'json',
       "type": "POST",
+      'async': false,
       "url": sSource,
       "data": aoData,
       "success": fnCallback
     });
   }
 });
+toggleFinancialSelections('#toggleSelect', 'assign');
```

Comment:
Disabling async in ajax calls is deprecated.
Add this call to your fnCallback instead and you don't need to do so. Like this:

    "success": function() {
      toggleFinancialSelections('#toggleSelect', 'assign');
      fnCallback();
    }